hw/syn/tools/dc/parse-syn-report.py

```diff
@@ -93,88 +93,145 @@ def _rel_err(val, ref):
-            results[key]["instances"][item[0]]["reg"] += float(item[4])
-            results[key]["instances"][item[0]]["macro"] += float(item[5])
-            results[key]["instances"][item[0]]["total"] += float(item[3]) + \
-                                                           float(item[4]) + \
-                                                           float(item[5])
-            comb_check += float(item[3])
-            reg_check += float(item[4])
-            macro_check += float(item[5])
+
+                # if we're not yet at depth, step one level down
+                # if this module has been specified
+                if inst[0] in args.expand_modules or depth < args.expand_depth:
+                    _extract_area_recursive(full_file,
+                                            results,
+                                            key,
+                                            args,
+                                            depth=depth + 1,
+                                            prefix=inst[0] + "/")
+
+            results[key]["instances"][inst[0]]["comb"] += float(inst[3])
+            results[key]["instances"][inst[0]]["reg"] += float(inst[4])
+            results[key]["instances"][inst[0]]["logic"] += float(inst[3]) + \
+                                                           float(inst[4])
+            results[key]["instances"][inst[0]]["macro"] += float(inst[5])
+            results[key]["instances"][inst[0]]["total"] += float(inst[3]) + \
+                                                           float(inst[4]) + \
+                                                           float(inst[5])
 
     except ValueError as err:
         results["messages"]["flow_errors"] += ["ValueError: %s" % err]
 
-    # cross check whether the above accounting is correct
-    if _rel_err(comb_check, results[key]["comb"]) > CROSSCHECK_REL_TOL:
-        results["messages"]["flow_errors"] += [
-            "Reporting error: comb_check (%e) != (%e)" %
-            (comb_check, results[key]["comb"])
-        ]
-    if _rel_err(reg_check, results[key]["reg"]) > CROSSCHECK_REL_TOL:
-        results["messages"]["flow_errors"] += [
-            "Reporting error: reg_check (%e) != (%e)" %
-            (reg_check, results[key]["reg"])
-        ]
-    if _rel_err(macro_check, results[key]["macro"]) > CROSSCHECK_REL_TOL:
-        results["messages"]["flow_errors"] += [
-            "Reporting error: macro_check (%e) != (%e)" %
-            (macro_check, results[key]["macro"])
-        ]
+    return
+
+
+def _check_area(results, key, args):
+    """
+    Checks whether the calculated area aggregates are
+    consistent among depth levels.
+    """
+
+    num_insts = len(results[key]["instances"])
+    instances = list(results[key]["instances"].values())
+    names = list(results[key]["instances"].keys())
+    for k, inst in enumerate(instances):
+        if k + 1 < num_insts:
```

Comment:
Couldn't you just iterate over `enumerate(instances[:-1])` instead, avoiding the `if` statement and a level of indentation?

Reply:
good idea, I'll change that